torch/testing/_internal/common_methods_invocations.py

```diff
@@ -262,9 +282,71 @@ def sample_inputs_addmm(self, device, dtype, requires_grad):
                                     low=None, high=None, 
                                     requires_grad=False))),) 
 
+# Metadata class for Fast Fourier Transforms in torch.fft.
+class SpectralFuncInfo(OpInfo):
+    """Operator information for torch.fft transforms. """
+
+    def __init__(self,
+                 name,  # the string name of the function
+                 *,
+                 ref=None,  # Reference implementation (probably in np.fft namespace)
+                 dtypes=floating_and_complex_types(),
+                 dtypesIfCPU=None,
+                 dtypesIfCUDA=None,
+                 dtypesIfROCM=None,
+                 ndimensional,
+                 skips=None,
+                 **kwargs):
+        skips = skips if skips is not None else []
+        dtypesIfCPU = dtypesIfCPU if dtypesIfCPU is not None else dtypes
+        dtypesIfCUDA = dtypesIfCUDA if dtypesIfCUDA is not None else dtypes
+        dtypesIfROCM = dtypesIfROCM if dtypesIfROCM is not None else dtypes
+
+        # gradgrad is quite slow
+        if not TEST_WITH_SLOW:
+            skips.append(SkipInfo('TestGradients', 'test_fn_gradgrad'))
+
+        if not TEST_MKL:
```

Comment:
These TEST_MKL and TEST_WITH_ROCM skips are cool but they can be applied universally using OpInfo's existing decorator support.

I think TEST_WITH_SLOW, however, does require special-casing like is done here, and this is probably the best way to do it.

Reply:
Isn't this better because it saves performing any checks when running the test in the non-skipped case.

Reply:
If we were worried about the performance impact you'd be correct, but I'd prefer we optimize for consistency/readability/maintainability here.

Reply:
Okay, I replaced the `SkipInfo`s with decorators.